albumentations/core/composition.py

```diff
@@ -114,8 +120,11 @@ def get_dict_with_id(self) -> Dict[str, Any]:
 
     def add_targets(self, additional_targets: Optional[Dict[str, str]]) -> None:
         if additional_targets:
+            self._additional_targets = {**self._additional_targets, **additional_targets}
```

Comment:
I think it would be great to check keys. And if they are exist, check if they are the same

Reply:
If we use this way - second dict keys overwrite keys from first. Second is after transforms. If no transforms applied to data - it will be same. And here we got references to same objects.

Reply:
Somethink like this:
```python
for k, v in additional_targets.items():
    if k in self._additional_targets and not v == self._additional_targets[k]:
        raise ValueError(f"Trying to owervwrite existed additional targets. Key={k} Exists={self._additional_targets[k]} New value: {v}")
    self._additional_targets[k] = v
```

Reply:
Ok, let's do it this way. It names `add`, not change...